cluster-operator/src/main/java/io/strimzi/operator/cluster/model/EntityTopicOperator.java

```diff
@@ -309,6 +319,38 @@ public RoleBinding generateRoleBinding(String namespace, String watchedNamespace
         return rb;
     }
 
+    @Override
+    protected String getServiceAccountRoleName() {
+        return EntityOperator.getRoleName(cluster);
+    }
+
+    public RoleBinding generateRoleBindingForRole(String namespace, String watchedNamespace) {
+        Subject ks = new SubjectBuilder()
+                .withKind("ServiceAccount")
+                .withName(EntityOperator.entityOperatorServiceAccountName(cluster))
+                .withNamespace(namespace)
+                .build();
+
+        RoleRef roleRef = new RoleRefBuilder()
+                .withName(getServiceAccountRoleName())
+                .withApiGroup("rbac.authorization.k8s.io")
+                .withKind("Role")
+                .build();
+
+        RoleBinding rb = new RoleBindingBuilder()
+                .withNewMetadata()
+                .withName(roleRoleBindingName(cluster))
+                .withNamespace(watchedNamespace)
+                .withOwnerReferences(createOwnerReference())
+                .withLabels(labels.toMap())
+                .endMetadata()
+                .withRoleRef(roleRef)
+                .withSubjects(singletonList(ks))
+                .build();
+
+        return rb;
+    }
```

Comment:
Could do, just following the same logic that was set for cluster roles, where the role bindings are handled separately for each operator container

Reply:
Actually, maybe that has to do with the difference between the watched namespace and the namespace they are running in? Maybe it needs to be separate in case each watches a separate namespace?